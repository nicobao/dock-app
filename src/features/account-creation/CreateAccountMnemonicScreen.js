import Clipboard from '@react-native-community/clipboard';
import {Button} from 'native-base';
import React, {useEffect} from 'react';
import {useSelector} from 'react-redux';
import {navigate} from '../../core/navigation';
import {Routes} from '../../core/routes';
import {showToast} from '../../core/toast';
import {
  Content,
  Footer,
  Header,
  LoadingButton,
  NBox as Box,
  ScreenContainer,
  Text,
  Theme,
  Typography,
} from '../../design-system';
import {BackButton} from '../../design-system';
import {translate} from '../../locales';
import {createAccountSelectors} from './create-account-slice';
import {withErrorBoundary} from '../../core/error-handler';
// TODO: Fix issues with Gradle 7
// import ScreenCaptureSecure from 'react-native-screen-capture-secure';

export function preventScreenShot() {
  // ScreenCaptureSecure.enableSecure();
}
export function enableScreenShot() {
  // ScreenCaptureSecure.disableSecure();
}
export function CreateAccountMnemonicScreen({phrase, onCopy, onSubmit}) {
  useEffect(() => {
    preventScreenShot();
    return () => {
      enableScreenShot();
    };
  }, []);
  return (
    <ScreenContainer testID="create-wallet-screen" hideGlobalHeader>
      <Header>
        <BackButton />
      </Header>
      <Content marginLeft={26} marginRight={26}>
        <Typography variant="h1" marginTop={52}>
          {translate('account_recovery_phrase.title')}
        </Typography>
        <Typography marginTop={12}>
          {translate('account_recovery_phrase.description')}
        </Typography>
        <Box
          flexDirection="row"
          flexWrap="wrap"
          backgroundColor={Theme.colors.secondaryBackground}
          borderRadius={12}
          p={5}
          mt={8}>
          {phrase.split(' ').map((word, idx) => (
            <Box flexDirection="row" width="33%" p={3} key={word}>
              <Text fontSize={14} mr={2} color={Theme.colors.description}>{`${
                idx + 1
              }`}</Text>
              <Text fontSize={14}>{word}</Text>
            </Box>
          ))}
        </Box>
        <Button
          full
          testID="next-btn"
          variant="unstyled"
          backgroundColor={Theme.colors.secondaryBackground}
          mt={5}
          onPress={onCopy}>
          {translate('account_recovery_phrase.copy_phrase')}
        </Button>
      </Content>
      <Footer marginBottom={0} marginLeft={26} marginRight={26}>
        <LoadingButton full testID="next-btn" mb={5} onPress={onSubmit}>
          {translate('navigation.next')}
        </LoadingButton>
      </Footer>
    </ScreenContainer>
  );
}

export const CreateAccountMnemonicContainer = withErrorBoundary(
  () => {
    const phrase = useSelector(createAccountSelectors.getMnemonicPhrase);

    const handleCopy = () => {
      Clipboard.setString(phrase);

      showToast({
        message: translate('account_recovery_phrase.phrase_copied'),
      });
    };

    const handleSubmit = () => navigate(Routes.CREATE_ACCOUNT_VERIFY_PHRASE);

    return (
      <CreateAccountMnemonicScreen
        phrase={phrase}
        onSubmit={handleSubmit}
        onCopy={handleCopy}
      />
    );
  },
  {disableLogs: true},
);
