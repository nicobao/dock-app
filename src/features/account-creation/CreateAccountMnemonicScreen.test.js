import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {CreateAccountMnemonicScreen} from './CreateAccountMnemonicScreen';
// TODO: Fix issues with Gradle 7
// import ScreenCaptureSecure from 'react-native-screen-capture-secure';

const mockStore = configureMockStore();

describe('CreateAccountMnemonicScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <CreateAccountMnemonicScreen phrase="test phrase" />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  // it('expect screenshot to be enabled', () => {
  //   enableScreenShot();
  //   expect(ScreenCaptureSecure.disableSecure).toBeCalled();
  // });

  // it('expect screenshot to be disabled', () => {
  //   preventScreenShot();
  //   expect(ScreenCaptureSecure.enableSecure).toBeCalled();
  // });
});
