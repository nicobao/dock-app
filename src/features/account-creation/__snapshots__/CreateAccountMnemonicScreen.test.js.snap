// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccountMnemonicScreen should render correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#141417",
      "flex": 1,
    }
  }
  testID="create-wallet-screen"
>
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <BackButton />
      </Styled(View)>
      <Styled(ScrollView)
        marginLeft={26}
        marginRight={26}
      >
        <Styled(Component)
          marginTop={52}
          variant="h1"
        >
          Account recovery phrase
        </Styled(Component)>
        <Styled(Component)
          marginTop={12}
        >
          Write this down or copy these words in the right order and save them somewhere safe. Do not share it to anyone
        </Styled(Component)>
        <Memo(ForwardRef(Box))
          backgroundColor="#27272A"
          borderRadius={12}
          flexDirection="row"
          flexWrap="wrap"
          mt={8}
          p={5}
        >
          <Memo(ForwardRef(Box))
            flexDirection="row"
            key="test"
            p={3}
            width="33%"
          >
            <Memo(ForwardRef(Text))
              color="#A1A1AA"
              fontSize={14}
              mr={2}
            >
              1
            </Memo(ForwardRef(Text))>
            <Memo(ForwardRef(Text))
              fontSize={14}
            >
              test
            </Memo(ForwardRef(Text))>
          </Memo(ForwardRef(Box))>
          <Memo(ForwardRef(Box))
            flexDirection="row"
            key="phrase"
            p={3}
            width="33%"
          >
            <Memo(ForwardRef(Text))
              color="#A1A1AA"
              fontSize={14}
              mr={2}
            >
              2
            </Memo(ForwardRef(Text))>
            <Memo(ForwardRef(Text))
              fontSize={14}
            >
              phrase
            </Memo(ForwardRef(Text))>
          </Memo(ForwardRef(Box))>
        </Memo(ForwardRef(Box))>
        <Memo(ForwardRef(Button))
          backgroundColor="#27272A"
          full={true}
          mt={5}
          testID="next-btn"
          variant="unstyled"
        >
          Copy to clipboard
        </Memo(ForwardRef(Button))>
      </Styled(ScrollView)>
      <Styled(Component)
        marginBottom={0}
        marginLeft={26}
        marginRight={26}
      >
        <LoadingButton
          full={true}
          mb={5}
          testID="next-btn"
        >
          Next
        </LoadingButton>
      </Styled(Component)>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;
